lib/run/measure.go

```diff
@@ -0,0 +1,116 @@
+package run
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/big"
+	"strconv"
+	"time"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+type record struct {
+	Height    *big.Int
+	BlockTime uint64
+	TxCount   uint64
+	GasLimit  uint64
+	GasUsed   uint64
+	PendingTxCount uint64
+}
+
+type txPoolStatus struct {
+        Pending string `json:"pending"` // Number of pending transactions
+        Queued  string `json:"queued"`  // Number of queued transactions
+}
+
+func MeasureTPS(rpcUrl string) {
+	client, err := ethclient.Dial(rpcUrl)
+	if err != nil {
+		log.Printf("Failed to connect to the Ethereum client: %v", err)
+		return
+	}
+
+	rpcClient, err := rpc.Dial(rpcUrl)
+	if err != nil {
+		log.Printf("Failed to connect to the Ethereum client: %v", err)
+		return
+	}
+
+        one := big.NewInt(1)
+
+        currentBlock, err := client.BlockByNumber(context.Background(), nil)
+        if err != nil {
+                log.Printf("Failed to get the start block: %v", err)
+		return
+        }
+
+        currentBlockHeight := currentBlock.Number()
```

Comment:
it should measure tps in historical blocks and calculate the max tps in a window (e.g. 1minute)